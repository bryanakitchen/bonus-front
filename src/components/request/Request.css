.Request {
    background: rgb(165, 184, 174);

    div {
        border: solid 1px black;
        padding: 4px;
        margin: 2px;
        background: white;
        border-radius: 5px;

        input {
            margin-left: 8px;
        }
    }
    
}