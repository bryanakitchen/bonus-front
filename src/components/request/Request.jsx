import React from 'react';
import PropTypes from 'prop-types';

function Request({ onSubmit, onChange, name, type, characteristic, method }) {

  return (
    <>
      <form onSubmit={onSubmit} >
        <h2>Insert an Animal</h2>
        <input 
          type="text" 
          placeholder="Enter name" 
          name="name"
          value={name}
          onChange={onChange} />
  
        <input 
          type="text" 
          placeholder="Enter type" 
          name="type"
          value={type}
          onChange={onChange} />
  
        <input 
          type="text" 
          placeholder="Enter characteristic" 
          name="characteristic"
          value={characteristic}
          onChange={onChange} />

        <br />
        <input 
          id="get"
          type="radio" 
          name="method" 
          value="get" 
          checked={method === 'get'}
          onChange={onChange} />
        <label htmlFor="get">Get all Animals</label>

        <input 
          id="post"
          type="radio" 
          name="method" 
          value="post"
          checked={method === 'post'}
          onChange={onChange} />
        <label htmlFor="post">Add an Animal</label>

        {/* <input 
          id="put"
          type="radio" 
          name="method" 
          value="put"
          onChange={onChange} />
        <label htmlFor="put">PUT</label> */}

        {/* <input 
          id="delete"
          type="radio" 
          name="method" 
          value="delete"
          checked={method === 'delete'}
          onChange={onChange} />
        <label htmlFor="delete">DELETE</label> */}
  
        <br />
        <button>Submit</button>
  
      </form>
    </>
  );
}

Request.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  characteristic: PropTypes.string.isRequired,
  method: PropTypes.string
};

export default Request;

