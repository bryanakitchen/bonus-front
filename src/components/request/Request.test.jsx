import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Request from './Request';

describe('Request component', () => {
  afterEach(() => cleanup());
  it('renders Request', () => {
    const { asFragment } = render(<Request 
      name="Tiger" 
      type="Mammal" 
      characteristic="stripes"
      onChange={jest.fn()}
      onSubmit={jest.fn()}  
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
