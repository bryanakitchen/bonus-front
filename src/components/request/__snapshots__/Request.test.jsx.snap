// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request component renders Request 1`] = `
<DocumentFragment>
  <form
    class="Request"
  >
    <h1>
      Animal Form
    </h1>
    <h3>
      Please select one
    </h3>
    <div>
      <input
        id="post"
        name="method"
        type="radio"
        value="post"
      />
      <label
        for="post"
      >
        Add an Animal
      </label>
      <input
        name="name"
        placeholder="Enter name"
        type="text"
        value="Tiger"
      />
      <input
        name="type"
        placeholder="Enter type"
        type="text"
        value="Mammal"
      />
      <input
        name="characteristic"
        placeholder="Enter characteristic"
        type="text"
        value="stripes"
      />
    </div>
    <div>
      <input
        id="get"
        name="method"
        type="radio"
        value="get"
      />
      <label
        for="get"
      >
        Get all Animals
      </label>
    </div>
    <div>
      <input
        id="find"
        name="method"
        type="radio"
        value="find"
      />
      <label
        for="find"
      >
        Find Animal by Id
      </label>
      <input
        name="id"
        placeholder="Enter ID of Animal"
        type="text"
        value=""
      />
    </div>
    <div>
      <input
        id="put"
        name="method"
        type="radio"
        value="put"
      />
      <label
        for="put"
      >
        Update Animal by Id
      </label>
      <input
        name="id"
        placeholder="Enter ID of Animal"
        type="text"
        value=""
      />
      <br />
      <input
        name="name"
        placeholder="Update name"
        type="text"
        value="Tiger"
      />
      <input
        name="type"
        placeholder="Update type"
        type="text"
        value="Mammal"
      />
      <input
        name="characteristic"
        placeholder="Update characteristic"
        type="text"
        value="stripes"
      />
    </div>
    <div>
      <input
        id="delete"
        name="method"
        type="radio"
        value="delete"
      />
      <label
        for="delete"
      >
        Delete Animal by Id
      </label>
      <input
        name="id"
        placeholder="Enter ID of Animal"
        type="text"
        value=""
      />
    </div>
    <button>
      Submit
    </button>
  </form>
</DocumentFragment>
`;
